pkg/client/record/event.go

```diff
@@ -195,10 +202,37 @@ func Event(object runtime.Object, reason, message string) {
 		Count:          1,
 	}
 
-	events.Action(watch.Added, e)
 }
 
 // Eventf is just like Event, but with Sprintf for the message field.
 func Eventf(object runtime.Object, reason, messageFmt string, args ...interface{}) {
 	Event(object, reason, fmt.Sprintf(messageFmt, args...))
 }
+
+type SourceInterface interface {
+	Event(object runtime.Object, reason, message string)
+	Eventf(object runtime.Object, reason, messageFmt string, args ...interface{})
+}
+
+func Source(source *api.EventSource) SourceInterface {
```

Comment:
It's messiest in the kubelet with the legacy config producing events.  That will get better once kubelet is using reflector.